core/common/src/main/java/alluxio/network/thrift/BootstrapTransport.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.network.thrift;
+
+import org.apache.thrift.transport.TTransport;
+import org.apache.thrift.transport.TTransportException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A transport that can support bootstrap.
+ */
+public abstract class BootstrapTransport extends TTransport {
```

Comment:
I see this class doing 2 things: (1) providing `isOpen`-checking functionality for operations and (2) being a convenient place to dump anything that is common between the subclasses that happen to extend it. I think we could make a clean abstract class around (1) which has no state but calls its subclass's `isOpen` method to guard operations. For (2), I don't think it's a good practice to mix so much between the base class and subclasses. If understanding a base class requires carefully examining its subclasses, something is wrong.

Reply:
separated into two concrete classes.